api/v1alpha4/virtualmachine_types.go

```diff
@@ -977,6 +1001,18 @@ type VirtualMachineStatus struct {
 	// TaskID describes the observed ID of the task created by VM Operator to
 	// perform some long-running operation on the VM.
 	TaskID string `json:"taskID,omitempty"`
+
+	// +optional
+
+	// CurrentSnapshot describes the observed working snapshot of the VirtualMachine.
+	CurrentSnapshot *corev1.TypedObjectReference `json:"currentSnapshot,omitempty"`
```

Comment:
do you want to use TypedLocalObjectReference here too ? any specific reason for using for non-namespaced (non-local) reference